pkg/kubelet/eviction/helpers.go

```diff
@@ -145,60 +144,34 @@ func ParseThresholdConfig(allocatableConfig []string, evictionHard, evictionSoft
 }
 
 // parseThresholdStatements parses the input statements into a list of Threshold objects.
-func parseThresholdStatements(expr string) ([]evictionapi.Threshold, error) {
-	if len(expr) == 0 {
+func parseThresholdStatements(statements map[string]string) ([]evictionapi.Threshold, error) {
+	if len(statements) == 0 {
 		return nil, nil
 	}
 	results := []evictionapi.Threshold{}
-	statements := strings.Split(expr, ",")
-	signalsFound := sets.NewString()
-	for _, statement := range statements {
-		result, err := parseThresholdStatement(statement)
+	for signal, val := range statements {
+		result, err := parseThresholdStatement(evictionapi.Signal(signal), val)
 		if err != nil {
 			return nil, err
 		}
-		if signalsFound.Has(string(result.Signal)) {
-			return nil, fmt.Errorf("found duplicate eviction threshold for signal %v", result.Signal)
```

Comment:
here we fail to parse the signal and return an error if we get a duplicate signal.

Reply:
Yeah, I'm aware. Now, you can't get any duplicates this deep because they get resolved by the flag parsing shim when the list is transformed to a map.

Reply:
Can we return an error when parsing the map to maintain this behavior?  Seems better to fail fast if someone inputs incorrect flags.

Reply:
Yeah, I suppose that's fine. Should be simple enough.

Reply:
I could do it for LangleSeparatedMapStringString, because that is only used on parameters for which we know there was a duplicate check, but I'd be worried about breaking backwards compatibility if I returned a duplicate key error from MapStringBool or MapStringString. MapStringBool is used for feature gates, which did not do a duplicate key check. MapStringString is used to replace the ConfigurationMap implementation that was living in kubeletconfig, which also did not do a duplicate key check.

Reply:
Probably best to be consistent across all flag shims.

Reply:
I agree consistency is good.  Is the plan then never to do duplicate checks, or always do them?

Reply:
Never for these flag shims, due to backwards-compatibility x consistency concerns.
Not sure whether the JSON or YAML parsers do duplicate checks.